pkg/commands/validate/validate.go

```diff
@@ -45,20 +43,23 @@ func action(ctx context.Context, fio files.FileReaderWriter, sio cmdlib.StdioRea
 	}
 }
 
-type componentValidator func([]*bundle.Component) field.ErrorList
-
-// componentValidationFn validates components.
-var componentValidationFn componentValidator = validate.AllComponents
 
 func runValidate(ctx context.Context, opts *options, brw cmdlib.BundleReaderWriter, gopt *cmdlib.GlobalOptions) error {
 	bw, err := brw.ReadBundleData(ctx, gopt)
 	if err != nil {
 		return fmt.Errorf("error reading contents: %v", err)
 	}
 
-	if errs := componentValidationFn(bw.AllComponents()); len(errs) > 0 {
-		return fmt.Errorf("there were one or more errors found while validating the bundle:\n%v", errs.ToAggregate())
+	bundleType := bw.Kind()
+	if bundleType == "Component" {
+		errs := validate.Component(bw.Component())
+		if len (errs) > 0 {
+			return fmt.Errorf("there were one or more errors found while validating the bundle:\n%v", errs.ToAggregate())
+		}
+	}else{   //@todo add validation for BundleBuilder, Bundle, ComponentBuilder here.
```

Comment:
Formatting's off here. Make sure that gofmt runs automatically.

Also, could you use a proper TODO?

// TODO(BradfordMedeiros): blah